### PR TITLE
Search for command using PATH from environment=.  Fixes #787

### LINKED ISSUE #787: Searching for command in PATH ignores changes made by environment=
I write a supervisord.conf，program section like this:

```
[program:xxx]
environment=PATH=%(ENV_HOME)s/.virtualenvs/venv/bin:%(ENV_PATH)s
command=gunicorn -w 3 xxx.app:app -b 0.0.0.0:8000
...
```

but it's fail. supervisor told me:

```
can't find command 'gunicorn'
```

but i can find it:

```
❯ ll ~/.virtualenvs/venv/bin/gunicorn 
-rwxrwxr-x 1 ubuntu ubuntu 245 Jun 12 05:39 /home/ubuntu/.virtualenvs/venv/bin/gunicorn
```

Does I used the wrong way?

### PR DESCRIPTION
**Pull request by dongweiming**
https://github.com/Supervisor/supervisor/pull/789
https://github.com/Supervisor/supervisor/issues/787

@mnaberez how about this?

### CODE REVIEW

**Comment by dongweiming** [comment]
@mnaberez

File: supervisor/options.py [hunk 1]
```diff
@@ -1481,13 +1481,17 @@ def _exit(self, code):
...
+        path = set(["/bin", "/usr/bin", "/usr/local/bin"])
+
         if "PATH" in os.environ:
             p = os.environ["PATH"]
             if p:
-                path = p.split(os.pathsep)
+                path = set(p.split(os.pathsep))
+
+        if env is not None and "PATH" in env:
+            path |= set(env["PATH"].split(os.pathsep))
```
Comment (mnaberez) [inline supervisor/options.py]: These unrelated changes from `list` to `set` need to be reverted.  A `set` is unordered so it is not appropriate for storing `PATH`.  It should remain a `list` as it was.

**Comment by mnaberez** [comment]
This patch changes an important part of how processes are spawned, but the changes have no tests.  I don't think we can merge changes to this code without adequate test coverage.

File: supervisor/options.py (hunk 1 above)
Comment (dongweiming) [inline supervisor/options.py]: ok fixed!

**Comment by dongweiming** [comment]
@mnaberez i has add two tests for this

**Comment by dongweiming** [comment]
@mnaberez pls review! thanks

**Comment by mnaberez** [comment]
@dongweiming Thanks for the change and added unit test!  I intend to include this patch in the next point release.  I will merge it after I have had a chance to test it locally.

**Comment by dongweiming** [comment]
@mnaberez yeah. But, what is your plan for release next version?

**Comment by mnaberez** [comment]
There's usually a point release every month or two ([changelog](http://supervisord.org/changes.html)).  We just had one so probably sometime next month.

**Comment by dongweiming** [comment]
@mnaberez got it

**Comment by dongweiming** [comment]
@mnabere What do you think of this？

**Comment by dongweiming** [comment]
@mnaberez still waiting on this

**Comment by dongweiming** [comment]
@mnaberez who about this pr?

**Comment by dongweiming** [comment]
If you won't approved it, I'll close it

**Comment by mnaberez** [comment]
Please leave it open, we have been busy with other things.  We intend to fix this.
